doc/decisions/1_in_discussion/spec_expressiveness.md

```diff
@@ -2,12 +2,38 @@
 
 ## Problem
 
-Currently, you can easily come into wrong assumptions that something would work in the specification.
-
-We need to find minimal requirements to implement a sane spec plugin.
+The current implementation of the spec plugin is copying metadata from the specification
+(`spec` namespace) to all other namespaces. By providing certain metadata e.g. 
+`type` or `description`, it can be assured that keys adhere to a given specification. 
+
+For some applications there are keys we don't want to define for every application explicitly.
+This is when `globbing` expressions in key names comes into place. By providing certain wildcard
+characters e.g. `_` or `#` one can match and specify a number of keys at once without the need
+to explicitly state the key in every specification.
+
+To describe the use case where `globbing` expressions make sense, we will use an application
+based on microservice architecture as sample.
+
+Imagine we have microservices and each of them communicates to a database to persist data. We
+will need some configuration so the applications can connect to the database e.g. `port`. Lets
+say we are specifying our configuration by `database/<DIALECT>/...`, where `...` is any name
+for a key. If we have multiple dialects e.g. a microservice uses more than one database e.g. `mysql`, `mssql`, etc. it is inconvenient to copy and paste the configuration for every 
+dialect we use.
+
+By using `globbing` expression we can overcome this problem.
+
+The key starts with `/sw/org/app-name`.
+```ni
+[database/_/port]
+meta:/require = true
+meta:/description = "The database port."
+```
+The `_` would match every `dialect` we are using in our application e.g. `postgresql`, `mssql`, etc.
```

Comment:
The example is a bit odd IMO, because it only makes sense, if one instance of the application uses multiple databases at the same time. If it just needs a single database, but lets the user choose which database the want to use, the specification could just be:

```ni
[database/port]
meta:/require = true
meta:/description = "The database port."

[database/dialect]
meta:/require = true
meta:/description = "The database dialect."
```

Maybe a better example, would be a service that acts as a reverse proxy:

```ni
[proxy/_/url]
meta:/require = true

[proxy/domain]
meta:/require = true
```

So this config

```
proxy/domain = example.com

proxy/bing/url = bing.com
proxy/google/url = google.com
```

would proxy requests for `bing.example.com` to `bing.com` and requests for `google.example.com` to `google.com`.

Or another example would be a mail client, which supports checking multiple accounts:

```ni
[account/#/email]
meta:/require = true

[account/#/password]
meta:/require = true
```